promoter/database.go

```diff
@@ -29,15 +33,37 @@ type (
 	operationType string
 
 	// updateFunc is the type of a function that can be used as a callback
-	// in threadedAddressWatcher.
-	updateFunc func(...WatchedAddressUpdate)
+	// in threadedAddressWatcher. Unused determines whether or not the
+	// 'unsed' flag is set in the API request for new addresses to watch.
+	// For deletions it will always be set to 'false'.
+	updateFunc func(unused bool, updates ...WatchedAddressUpdate) error
+
+	// User is the type of a user in the database.
+	// TODO: f/u with API endpoint for creating users and assigning
+	// addresses.
+	User struct {
+		ID  primitive.ObjectID `bson:"_id"`
+		Sub string             `bson:"sub"`
```

Comment:
I'd rather not unless we think there is a clear need for it. Because I'm not a fan of duplicating state which might diverge. Especially across collection. 

A potentially better solution would be to just make the Sub the `_id`. Then we can directly use the Sub everywhere without any cross-collection magic or lookups.

Reply:
What is the state you don't want to duplicate?

> A potentially better solution would be to just make the Sub the _id

Reply:
The watched addresses are already in their own collection and each has a userid. So I don't want to have each user's addresses stored within the user as a set as well. Because we might end up with addresses being deleted from one of them but still existing in the other, assigned to the user.

I have seen lots of Go examples using a `String` as the ID so I don't think it's that big of a deal. Seems like any type apart from an array is allowed.